src/matrix/lkj.jl

```diff
@@ -186,14 +186,16 @@ end
 #  for 1 / c₀, even if they say that it is not.
 #  -----------------------------------------------------------------------------
 
-function lkj_onion_loginvconst(d::Integer, η::Real)
+function lkj_onion_loginvconst(d::Integer, η::T) where {T <: Real}
     #  Equation (17) in LKJ (2009 JMA)
-    sumlogs = zero(η)
-    for k in 2:d - 1
-        sumlogs += 0.5k*logπ + loggamma(η + 0.5(d - 1 - k))
+    h = convert(float(T), 0.5)
+    sumlogs = logπ + loggamma(η + h * (d - 3))
+    sumlogs = ifelse(d >= 3, sumlogs, zero(sumlogs))
+    for k in 3:d - 1
+        sumlogs += (h * k) * logπ + loggamma(η + h * (d - 1 - k))
```

Comment:
As I wrote in my previous comment, I think it would be simpler to just initialize with the other terms in `loginvconst` below. Moreover, it would be more efficient to move the `logπ` computation out of the loop. With the definition of `T` suggested above one could write
```suggestion
    α = η + h * d - 1
    loginvconst = (2*η + d - 3)*oftype(T, logtwo) + (oftype(T, logπ) / 4) * (d * (d - 1) - 2) + logbeta(α, α) - (d - 2) * loggamma(η + h * (d - 1))
    for k in 2:(d - 1)
        loginvconst += loggamma(η + h * (d - 1 - k))
```